bundles/org.openhab.binding.hive/src/main/java/org/openhab/binding/hive/internal/client/BuilderUtil.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hive.internal.client;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ *
+ *
+ * @author Ross Brown - Initial contribution
+ */
+@NonNullByDefault
+public final class BuilderUtil {
+    public static final String REQUIRED_ATTRIBUTE_NOT_SET_MESSAGE = "Required attributes have not been set";
+
+    private BuilderUtil() {
+        throw new AssertionError();
```

Comment:
I did this following the style shown in Effective Java by Joshua Bloch.
He says:
> The AssertionError isn't strictly required, but it provides insurance in case the constructor is accidentally invoked from within the class.

N.B. If you think I should remove this I have also used this idiom in:
- HiveBindingConstants
- FeatureAttributeFactory
- HiveApiConstants
- TestUtil

Reply:
Well I don't think you necessarily need to remove it, I just feel that an error is a little overkill since even if an instance of that class was constructor no harm would come of it. So you are turning a normally-impossible non-problem into a normally-impossible big problem.

Just my 2 cents.

Reply:
Just adding another 2 cents:

This kind of defensive programming is very appropriate for library development where it is possible for users of your library to poke and prod at your classes outside of the apis that you provide them. But bindings aren't libraries, they are plugins, which means that you know and control exactly where your code is poked. Being overly defensive in plugin code only adds bloat. Defensive coding styles are usually the best approach, but not so much in this case.

Reply:
I have been writing this with the view that everything inside the `client` package is a library because it might be useful in other projects. I originally had it as a separate project but decided to bundle it in and leave learning about publishing to maven central/jcenter for another day.

I realise in the process of playing around with this project I have probably made things far more bloated than they need to be.

I feel a bit silly comparing the previous binding with +1,365 lines vs my in comparison monstrous +16,517.